lib/statsd.js

```diff
@@ -402,7 +348,11 @@ Client.prototype.sendMessage = function (message, callback) {
     return;
   }
 
-  if (this.protocol === TCP && message.lastIndexOf('\n') !== message.length - 1) {
+  if (!this.socket) {
+    return callback(new Error('No transport set. Check previous errors for details.'));
+  }
+
+  if (this.protocol === PROTOCOL.TCP && message.lastIndexOf('\n') !== message.length - 1) {
     message += '\n';
   }
```

Comment:
I think this would be a better fit inside lib/transport.js

Reply:
Good point!